### ISSUE TITLE
handle "argument list too long"

### ISSUE THREAD

**Issue by kzidane** [cs50/lib50#40]
When trying to submit too many files (e.g., their `node_modules` or virtual environment files). Ideally these files should be ignored but they shouldn't cause our tools to break if they're not.

**Comment by crossroads1112** [comment]
This is an OS-imposed limit. Things like `rm`, `cp`, etc. will break too for the same reason. We could let patterns be specified as strings and glob them ourselves I guess (though the number of patterns is still limited of course)

**Comment by kzidane** [comment]
Yeah, I understand that. What I had in mind was detect whatever error message `git` shows up in this case and show a more friendly output with a hint or two maybe instead of saying "something went wrong". Alternatively, maybe we could run `git add` once per listing though that might end up being significantly slower.

**Comment by crossroads1112** [comment]
Oh that's where the error is showing up, right. I forgot you can't specify files manually to check/submit50 anymore.

It's way way slower to run git add manually for each file even for something like finance. I remember making that change.

Hmm... I wonder if we can get that value dynamically and do it in batches. Otherwise we could find the lowest number that reasonable platforms allow and use that.
